File: test_c/test_fibrec.c
#include <stdio.h>

long long fibrec(int n, long long *opC) {
    if (n <= 2) { return 1; }
    long long x = fibrec(n-1, opC) + fibrec(n-2, opC);
    (*opC)++;
    return x;
}

int main() {
    long long opC = 0;
    int passed = 0;
    if (fibrec(1, &opC) == 1)    { printf("test 1 passed\n"); passed++; }
    if (fibrec(2, &opC) == 1)    { printf("test 2 passed\n"); passed++; }
    if (fibrec(7, &opC) == 13)   { printf("test 3 passed\n"); passed++; }
    if (fibrec(10, &opC) == 55)  { printf("test 4 passed\n"); passed++; }
    if (fibrec(20, &opC) == 6765){ printf("test 5 passed\n"); passed++; }
    printf("%d/5 tests passed.\n", passed);
    return 0;
}

File: test_rs/test_fibrec.rs
fn fibrec(n: u64, op_c: &mut u64) -> u64 {
    if n <= 2 { return 1; }
    let x: u64 = fibrec(n-1, op_c) + fibrec(n-2, op_c);
    *op_c += 1;
    x
}

fn main() {
    let mut op_c: u64 = 0;
    let mut passed = 0;
    if fibrec(1, &mut op_c) == 1    { println!("test 1 passed"); passed += 1; }
    if fibrec(2, &mut op_c) == 1    { println!("test 2 passed"); passed += 1; }
    if fibrec(7, &mut op_c) == 13   { println!("test 3 passed"); passed += 1; }
    if fibrec(10, &mut op_c) == 55  { println!("test 4 passed"); passed += 1; }
    if fibrec(20, &mut op_c) == 6765{ println!("test 5 passed"); passed += 1; }
    println!("{}/5 tests passed.", passed);
}
